src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Avalonia.Controls.Templates;
+using Avalonia.Data.Converters;
+
+namespace Avalonia.Controls.Converters;
+
+/// <summary>
+/// Select a valid IDataTemplate in order.
+/// </summary>
+public class MultiDataTemplatesConverter: IMultiValueConverter
```

Comment:
I am not sure about multibinding approach. It makes whole feature highly dependent on a template, and well as adds a very specific converter to the public API.

Reply:
Yes, I think  this is over designed. Maybe we can just remove ContentTemplate local value from ComboBox template, and set it in root style?

Reply:
You could also use the coercion callback to coerce this new property to `ItemTemplate` when it's null. This should simplify the XAML side with a classic single `{TemplateBinding}`.